src/Symfony/Component/DependencyInjection/Definition.php

```diff
@@ -830,6 +831,38 @@ public function isAbstract()
     }
 
     /**
+     * Whether this definition is deprecated, that means it should not be called
+     * anymore
+     *
+     * @param bool $boolean
+     *
+     * @return Definition the current instance
+     *
+     * @todo Support a suggested definition instead ?
+     *
+     * @api
+     */
+    public function setDeprecated($boolean = true)
+    {
+        $this->deprecated = (bool) $boolean;
+
+        return $this;
+    }
+
+    /**
+     * Whether this definition is deprecated, that means it should not be called
+     * anymore
+     *
+     * @return bool
+     *
+     * @api
```

Comment:
I ported the tags from the abstract behaviour, but I can remove those if it's too soon to put them here